src/GraphQl/Serializer/ItemNormalizer.php

```diff
@@ -110,6 +111,25 @@ public function normalize(mixed $object, string $format = null, array $context =
      */
     protected function normalizeCollectionOfRelations(ApiProperty $propertyMetadata, iterable $attributeValue, string $resourceClass, ?string $format, array $context): array
     {
+        if ('item_query' === $context['graphql_operation_name'] && $this->resourceMetadataCollectionFactory) {
+            // check for nested collection
+            $operation = $this->resourceMetadataCollectionFactory->create($resourceClass)->getOperation('collection_query');
+            if ($operation instanceof Query && $operation->getNested() && !$operation->getResolver() && !$operation->getProvider()) {
+                if (false === $operation->getPaginationEnabled()) {
+                    return [...$attributeValue];
+                }
+
+                $data = ['edges' => []];
```

Comment:
I'm not sure to understand why the serialization of the edges / nodes is done here. It's already done in the `NormalizeProcessor`, why doesn't it pass through it?

Reply:
Answer to myself: probably because it's not a resource but a JSON array. However that means we lose what is done in the processor (page information for instance) or we need to duplicate the logic.
IMO we need to find a solution before merging this PR.

Reply:
Maybe we should go one step back and discuss the scope:

As I mentioned in the description of this PR, this feature is to simplify resolving a nested collection and treat them more like **properties** than relations. The consequence was that also the edges / nodes representation was not in my mind, because this is not for properties. I first thought it is easy to have also the edges / nodes support as you demanded and have both opportunities (controlled via `paginationEnabled` flag). But maybe we should not provide that for this case. My proposal would be to revert the latest changes.

Would like to hear your thoughts @alanpoulain.

Reply:
I don't see why it should be treated like properties and not like a collection. It should have all the possibilities of a classical relation.

Reply:
Found my mistake. I started creating this feature on 3.2 and added a special case to the `CollectionResolverFactory`. On the main branch the general `ResolverFactory` is already used. So I added the special case also there and voilà: The changes in `ItemNormalizer` could have been reverted. Thx for your support.